### PR TITLE
feat(journey-maps): horizontal level switch

### PR DESCRIPTION
**Pull request by swiss-chris**
https://flow.sbb.ch/browse/ROKAS-1723

image

### CODE REVIEW

File: src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SbbLevelSwitchHorizontal implements OnDestroy {
+  @Input() map: MaplibreMap | null;
+  @Input() showSmallButtons: boolean | undefined;
+
+  @ViewChild('mainButton') mainButton: ElementRef<HTMLButtonElement>;
+  @ViewChildren('sideButton') sideButtons: QueryList<ElementRef<HTMLButtonElement>>;
+  showSideButtons: boolean = false;
+  private countdownTimer: ReturnType<typeof setTimeout>;
```
Comment (mstankala) [inline src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts:43]:
```suggestion
  private _countdownTimer: ReturnType<typeof setTimeout>;
```

File: src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  private autoCollapseTimeout = 5000; // 5000 ms
```
Comment (mstankala) [inline src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts:44]:
```suggestion
  private _autoCollapseTimeout = 5000; // 5000 ms
```

File: src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      this.startCountdown();
+      setTimeout(() => {
+        this.focusMatchingButton();
+      }, 0); // make sure side buttons are visible first
+    } else {
+      clearTimeout(this.countdownTimer);
+    }
+  }
+
+  private focusMatchingButton(): void {
```
Comment (mstankala) [inline src/journey-maps/angular/components/level-switch/level-switch-horizontal.ts:70]:
```suggestion
  private _focusMatchingButton(): void {
```

**Comment by mstankala** [comment]
I feel like the animation when unfolding is different when a level is selected and when not. Could that be? 
When no level is selected, it looks very good. 
When a level is selected, there seems to be an overlay when unfolding.

**Review by mstankala — CHANGES_REQUESTED** [review]
It looks, like the animation when unfolding is different when a level is selected and when not. Could that be?
When no level is selected, it looks very good.
When a level is selected, there seems to be an overlay when unfolding.

**Review by mstankala — APPROVED** [review]
LGTM

**Review by mhaertwig — APPROVED** [review]
LGTM 👍

**Pull request merged**